snapshot/lib/snapshot/runner.rb

```diff
@@ -204,6 +208,20 @@ def erase_simulator(device_type)
       `xcrun simctl erase #{device_udid} &> /dev/null`
     end
 
+    def localize_simulator(device_type, language, locale)
+      device_udid = TestCommandGenerator.device_udid(device_type)
+      if device_udid
+        locale ||= language.sub("-", "_")
+        plist = {
+          AppleLocale: locale,
+          AppleLanguages: [language]
+        }
+        UI.message "Localizing #{device_type} (AppleLocale=#{locale} AppleLanguages=[#{language}])"
+        plist_path = "#{ENV['HOME']}/Library/Developer/CoreSimulator/Devices/#{device_udid}/data/Library/Preferences/.GlobalPreferences.plist"
+        File.write(plist_path, Plist::Emit.dump(plist))
```

Comment:
`File.write` will indeed fail if the whole directory path is not present.

A defensive `FileUtils.mkdir_p` first for the directory path would make sense unless that path is otherwise guaranteed to exist.